heartbeat/ocf-distro

```diff
@@ -5,43 +5,205 @@
 # released RH specific stuff (/etc/sysconfig/nfs editing)
 # 
 
+# systemd-based systems should all have an os-release file.
+_ETC_OS_RELEASE_FILE="/etc/os-release"
+_USR_OS_RELEASE_FILE="/usr/lib/os-release"
+
+# Legacy distro-specific files
 _DEBIAN_VERSION_FILE="/etc/debian_version"
 _REDHAT_RELEASE_FILE="/etc/redhat-release"
 _SUSE_RELEASE_FILE="/etc/SuSE-release"
-_RELEASE_FILES="/etc/*-release"
-_REDHAT_BASED_DISTROS_RE='red *hat|fedora|centos|scientific linux'
 
-get_release_id() {
+_DEBIAN_BASED_DISTROS_RE="debian|ubuntu"
+_REDHAT_BASED_DISTROS_RE="red *hat|rhel|fedora|centos|scientific"
+_SUSE_BASED_DISTROS_RE="sles|suse"
+
+
+# Convert OS release ID to a standard form regardless of source.
+_process_os_release_id() {
+	_os="$1"
+
+	# Isolate distro name, remove whitespace, convert to lowercase
+	_os=$(echo "$_os" | sed -e "s/\([G][Nn][Uu]\/\)\?[Ll][Ii][Nn][Uu][Xx]//" \
```

Comment:
`/i` isn't POSIX-compliant, so it may throw an error on BSD. As it turns out, neither is the use of `\?` (which just gets treated as a literal), so I'll need to remove that too.
```
# _os="Red Hat Enterprise Linux"
# echo "$_os" | sed --posix -e "s/\([G][Nn][Uu]\/\)\?[Ll][Ii][Nn][Uu][Xx]//"
Red Hat Enterprise Linux

# echo "$_os" | sed -e "s/\([G][Nn][Uu]\/\)\?[Ll][Ii][Nn][Uu][Xx]//"
Red Hat Enterprise 

# echo "$_os" | sed --posix -e "s/Linux//i"
sed: -e expression #1, char 10: unknown option to `s'

# echo "$_os" | sed -e "s/Linux//i"
Red Hat Enterprise 
```